lib/puma/server.rb

```diff
@@ -595,6 +600,24 @@ def handle_request(req, lines)
       env[RACK_INPUT] = body
       env[RACK_URL_SCHEME] =  env[HTTPS_KEY] ? HTTPS : HTTP
 
+      if @early_hints
+        env[EARLY_HINTS] = lambda { |links|
```

Comment:
It would be nice to have this lambda live somewhere other than the body of this method.

Reply:
Agree.  @eileencodes and I were trying to move it out of here, but the lambda body needs to access the io object (`client`) and `fast_write` is a private method.  I'd love to move the lambda out of here, but I am not sure how.